store/acitreestore/tree.go

```diff
@@ -104,20 +242,26 @@ func (ts *TreeStore) Write(id string, key string, s *Store) (string, error) {
 		return "", errwrap.Wrap(errors.New("failed to sync tree store directory"), err)
 	}
 
+	// TODO(sgotti) this is wrong for various reasons:
```

Comment:
> Is it a bigger problem?

If we want to create an aci store interface to be used by multiple implementations then these methods looks too much specialized. I shared some ideas and other thoughts in this #2890. The basic idea will be to save the tree store sizes inside the tree store and sum it up at `rkt image list` time, to avoid O(n) tree store walks it'll need a new implementation backed by some indexes (like a ql db) and this will need a way to handle a seamless migration.


> Do we have to submit an issue for this?

I wanted to hear some thoughts on #2890 . If you prefer I'll open it with the specific problem.